src/averageAges.js

```diff
@@ -55,7 +86,31 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const foundMan = people.filter(man => man.sex === 'm');
+  let age = 0;
+  let count = 0;
+
+  if (onlyWithSon) {
+    foundMan.forEach(man => {
+      people.forEach(woman => {
+        if (man.mother === woman.name) {
+          age += (man.born - woman.born);
+          count++;
+        }
+      });
+    });
```

Comment:
Same here, you use the same logic for both of variants
Just set correct array for some variable and use it in your code